deck.rb

```diff
@@ -0,0 +1,32 @@
+$LOAD_PATH << '.'
+require 'card'
+
+# Class structure for Deck, which is a collection of cards
+class Deck
+  attr_reader :game_deck
+
+  def populate_deck(nums, suits)
+    # Populates deck with cards of each num and each suit
+    nums.each do |num|
```

Comment:
Idiomatic ruby tends to use the functional iterators nowadays, so rather than shoveling inside a loop you can assign from the outside via map/flatten:

```ruby
@game_deck = nums.map { |num| suits.map { |suit| Card.new(num, suit) } }.flatten
```

`map` returns an array with each return value of the block, so this gives you a nested array of cards by number, without side effects. `flatten` then collapses the per-number arrays all into one.